src/main/java/htsjdk/samtools/util/IOUtil.java

```diff
@@ -280,6 +301,79 @@ public static File getDefaultTmpDir() {
         else return new File(tmp, user);
     }
 
+    /**
+     * Creates a new tmp path on one of the available temp filesystems, registers it for deletion
+     * on JVM exit and then returns it.
+     */
+    public static Path newTempPath(final String prefix, final String suffix,
+            final Path[] tmpDirs, final long minBytesFree) throws IOException {
+        Path p = null;
+
+        for (int i = 0; i < tmpDirs.length; ++i) {
+            if (i == tmpDirs.length - 1 || Files.getFileStore(tmpDirs[i]).getUsableSpace() > minBytesFree) {
+                p = Files.createTempFile(tmpDirs[i], prefix, suffix);
+                deleteOnExit(p);
+                break;
+            }
+        }
+
+        return p;
+    }
+
+    /** Creates a new tmp file on one of the potential filesystems that has at least 5GB free. */
+    public static Path newTempPath(final String prefix, final String suffix,
+            final Path[] tmpDirs) throws IOException {
+        return newTempPath(prefix, suffix, tmpDirs, FIVE_GBS);
+    }
+
+    /** Returns a default tmp directory as a Path. */
+    public static Path getDefaultTmpDirPath() {
+        try {
+            final String user = System.getProperty("user.name");
+            final String tmp = System.getProperty("java.io.tmpdir");
+
+            final Path tmpParent = getPath(tmp);
+            if (tmpParent.endsWith(tmpParent.getFileSystem().getSeparator() + user)) {
+                return tmpParent;
```

Comment:
That requires to change the properties within the test - I am not sure if that could have side-effects in other tests. Any idea?

Reply:
This method is retrieving the default temp directory, which comes from the java properties (`java.io.tmpdir` and `user.name`). If the `java.io.tmpdir` property already provides the user folder, it returns this directly; otherwise, it resolves the path for the `java.io.tmpdir`/`user.name`.

The test `IoUtilsTest.testHetDefaultTmpDirPath()` (misnamed, so changed in the next commit), is testing the behaviour for the java-properties in the system; if we want to test all the codepaths we should change the properties there. I do not know if this will have side effects in other tests. I will try to change it in an independent commit to be able to remove it if so.